src/Module/Profile/Photos/Upload.php

```diff
@@ -0,0 +1,277 @@
+					$filesize = intval($_FILES['media']['size'][0]);
+				} else {
+					$filesize = intval($_FILES['media']['size']);
+				}
+			}
+
+			if (!empty($_FILES['media']['type'])) {
+				if (is_array($_FILES['media']['type'])) {
+					$filetype = $_FILES['media']['type'][0];
+				} else {
+					$filetype = $_FILES['media']['type'];
+				}
+			}
+		}
+
+		if ($src == '') {
+			$this->logger->warning('File source (temporary file) cannot be determined', ['$_FILES' => $_FILES]);
+			$this->exit(401, $this->l10n->t('Invalid request.'), true);
+		}
+
+		$filetype = Images::getMimeTypeBySource($src, $filename, $filetype);
+
+		$this->logger->info('File upload:', [
+			'src'      => $src,
+			'filename' => $filename,
+			'filesize' => $filesize,
+			'filetype' => $filetype,
+		]);
+
+		$imagedata = @file_get_contents($src);
+		$image     = new Image($imagedata, $filetype);
+
+		if (!$image->isValid()) {
+			@unlink($src);
+			$this->logger->warning($this->l10n->t('Unable to process image.'), ['imagedata[]' => gettype($imagedata), 'filetype' => $filetype]);
+			$this->exit(401, $this->l10n->t('Unable to process image.'));
+		}
+
+		$image->orient($src);
+		@unlink($src);
+
+		$max_length = $this->config->get('system', 'max_image_length');
+		if ($max_length > 0) {
+			$image->scaleDown($max_length);
+			$filesize = strlen($image->asString());
+			$this->logger->info('File upload: Scaling picture to new size', ['max_length' => $max_length]);
+		}
+
+		$width  = $image->getWidth();
+		$height = $image->getHeight();
+
+		$maximagesize = $this->config->get('system', 'maximagesize');
+
+		if (!empty($maximagesize) && $filesize > $maximagesize) {
+			// Scale down to multiples of 640 until the maximum size isn't exceeded anymore
+			foreach ([5120, 2560, 1280, 640] as $pixels) {
+				if ($filesize > $maximagesize && max($width, $height) > $pixels) {
+					$this->logger->info('Resize', ['size' => $filesize, 'width' => $width, 'height' => $height, 'max' => $maximagesize, 'pixels' => $pixels]);
+					$image->scaleDown($pixels);
+					$filesize = strlen($image->asString());
+					$width    = $image->getWidth();
+					$height   = $image->getHeight();
+				}
+			}
+
+			if ($filesize > $maximagesize) {
+				@unlink($src);
+				$this->logger->notice('Image size is too big', ['size' => $filesize, 'max' => $maximagesize]);
+				$this->exit(401, $this->l10n->t('Image exceeds size limit of %s', Strings::formatBytes($maximagesize)));
+			}
+		}
+
+		$resource_id = Photo::newResource();
+
+		$smallest = 0;
+
+		// If we don't have an album name use the Wall Photos album
+		if (!strlen($album)) {
+			$album = $this->l10n->t('Wall Photos');
+		}
+
+		$defperm = '<' . $default_cid . '>';
+
+		$result = Photo::store($image, $page_owner_uid, $visitor, $resource_id, $filename, $album, 0, Photo::DEFAULT, $defperm);
+		if (!$result) {
+			$this->logger->warning('Photo::store() failed', ['result' => $result]);
+			$this->exit(401, $this->l10n->t('Image upload failed.'));
```

Comment:
```suggestion
			$this->exit(401, $this->t('Image upload failed.'));
```

Reply:
Right, I always forget there are BaseModule wrappers for these.